provider/datagen/src/transform/cldr/units/info.rs

```diff
@@ -0,0 +1,166 @@
+                Some(ref offset) => offset.as_str(),
+                None => "0",
+            };
+
+            let factor_scientific = process_factor(factor, &clean_constants_map)?;
+            let offset_scientific = process_factor(offset, &clean_constants_map)?;
+
+            let convert_unit_index = convert_units_vec.len();
+            convert_units_vec.push(extract_conversion_info(
+                base_unit,
+                factor_scientific,
+                offset_scientific,
+            )?);
+
+            if let Some(units_info_index) = conversion_info_map.get_mut(unit_name.as_str()) {
+                units_info_index.convert_info = Some(convert_unit_index as u16);
+            } else {
+                let units_info_index = UnitsInfoIndex {
+                    dimension: None,
+                    convert_info: Some(convert_unit_index as u16),
+                };
+                conversion_info_map.insert(unit_name.as_str(), units_info_index);
+            }
+        }
+
+        let result = UnitsInfoV1 {
+            units_info: ZeroMap::from_iter(conversion_info_map),
+            unit_dimensions: VarZeroVec::from(&quantity_vec),
+            convert_infos: VarZeroVec::from(&convert_units_vec),
+        };
+
+        Ok(DataResponse {
+            metadata: Default::default(),
+            payload: Some(DataPayload::from_owned(result)),
+        })
+    }
+}
+
+impl IterableDataProvider<UnitsInfoV1Marker> for crate::DatagenProvider {
+    fn supported_locales(&self) -> Result<Vec<DataLocale>, DataError> {
+        Ok(vec![Default::default()])
+    }
+}
+
+#[test]
+fn test_basic() {
+    use icu_locid::locale;
+    use icu_provider::prelude::*;
+    use icu_unitsconversion::provider::Sign;
+    use num_bigint::BigUint;
+    use zerovec::maps::ZeroVecLike;
+
+    let provider = crate::DatagenProvider::new_testing();
+
+    let und: DataPayload<UnitsInfoV1Marker> = provider
+        .load(DataRequest {
+            locale: &locale!("und").into(),
+            metadata: Default::default(),
+        })
+        .unwrap()
+        .take_payload()
+        .unwrap();
+
+    let units_info = und.get().to_owned();
+    let units_info_map = &units_info.units_info;
+    let unit_quantity = &units_info.unit_dimensions;
+    let convert_units = &units_info.convert_infos;
+
+    let meter = units_info_map.get("meter").unwrap();
+    let meter_quantity_index = meter.dimension.get().unwrap().as_unsigned_int() as usize;
+    let meter_quantity = unit_quantity.zvl_get(meter_quantity_index).unwrap();
+    assert_eq!(&meter_quantity.quantity, "length");
+    // TODO: how to test this?
+    // assert_eq!(meter_quantity.constant_exactness as u8, QuantitySimplicity::Simple as u8);
+
+    let big_one = BigUint::from(1u32);
+
+    let meter_convert_index = meter.convert_info.get().unwrap().as_unsigned_int() as usize;
+    let meter_convert_ule = convert_units.zvl_get(meter_convert_index).unwrap();
```

Comment:
@sffc and @Manishearth 

I would like to convert from `ConversionInfoULE` to `ConversionInfo`.

I have tried `ConversionInfo::from_unaligned(ule)` as @Manishearth  suggested in the previous comment, but I could not find it.

Reply:
It's a VarULE type. Use ZeroFrom::zero_from

Reply:
Great, it worked. thanks :)